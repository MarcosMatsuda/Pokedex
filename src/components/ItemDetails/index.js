// import React from 'react';
// import { SafeAreaView, Text, FlatList, View } from 'react-native';
// import { useSelector } from 'react-redux';
// import { useNavigation } from '@react-navigation/native';

// import { ProgressiveDiscount } from '~/components';

// import { Container } from './styles';

// const ItemDetails = ({ title }) => (
//   <View>
//     <Text>{title}</Text>
//   </View>
// );

// export { ItemDetails };
