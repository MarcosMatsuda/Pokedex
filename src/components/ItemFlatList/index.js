import React from 'react';
import { useNavigation } from '@react-navigation/native';

import { useDispatch, useSelector } from 'react-redux';
import { addTofavorites, removeFromFavorites } from '../../store/ducks/pokedex';

import {
  Container,
  Title,
  ButtonContainer,
  Star,
  DetailContainer,
} from './styles';

const ItemFlatList = (props) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const {
    pokemon: { name, favorite },
  } = props;

  const pokemons = useSelector((state) => state.pokedex.pokemons);

  let isFavorite =
    favorite === undefined
      ? !!pokemons.find((el) => el.name === name)
      : favorite;

  return (
    <Container>
      {isFavorite ? (
        <ButtonContainer onPress={() => dispatch(removeFromFavorites(name))}>
          <Star name="star" />
        </ButtonContainer>
      ) : (
        <ButtonContainer onPress={() => dispatch(addTofavorites(name))}>
          <Star name="star-outline" />
        </ButtonContainer>
      )}
      <DetailContainer
        onPress={() =>
          navigation.navigate('Details', {
            name,
          })
        }>
        <Title>{name}</Title>
      </DetailContainer>
    </Container>
  );
};

export { ItemFlatList };
