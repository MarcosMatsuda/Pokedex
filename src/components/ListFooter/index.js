import React from 'react';
import { ActivityIndicator } from 'react-native';
import { Container } from './styles';
const ListFooter = ({ loading }) => {
  if (!loading) {
    return null;
  }
  return (
    <Container>
      <ActivityIndicator animating size="large" />
    </Container>
  );
};

export { ListFooter };
