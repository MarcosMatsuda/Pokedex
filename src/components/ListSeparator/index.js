import React from 'react';
import { Separator } from './styles';

const ListSeparator = () => {
  return <Separator />;
};

export { ListSeparator };
