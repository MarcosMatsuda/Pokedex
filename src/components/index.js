export { ItemFlatList } from './ItemFlatList';
export { ListSeparator } from './ListSeparator';
export { ListFooter } from './ListFooter';
