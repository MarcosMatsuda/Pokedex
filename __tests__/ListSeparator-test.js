import 'react-native';
import React from 'react';
import { ListSeparator } from '~/components';
import renderer from 'react-test-renderer';

describe('Snapshot testing', () => {
  test('should take Snapshot', () => {
    const tree = renderer.create(<ListSeparator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
