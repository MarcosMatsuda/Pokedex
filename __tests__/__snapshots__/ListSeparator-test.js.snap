// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing should take Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#CED0CE",
      "height": 1,
      "width": "100%",
    }
  }
/>
`;
